src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/TemplatingPass.php

```diff
@@ -33,6 +36,8 @@ public function process(ContainerBuilder $container)
         if ($container->hasDefinition('templating.engine.php')) {
             $helpers = array();
             foreach ($container->findTaggedServiceIds('templating.helper') as $id => $attributes) {
+                @trigger_error('The '.self::class.' class is deprecated since version 3.3 and will be removed in 4.0. Use Twig instead.', E_USER_DEPRECATED);
```

Comment:
Wouldn't it be even better to have this deprecation only when using the class itself and trigger something like `The "templating.helper" tag is deprecated as of 3.3...`  here instead?